### PR TITLE
Add cluster healthcheck and take backups from most advanced node

### PR DESCRIPTION
**Pull request by galal-hussein**
https://github.com/rancher/rancher/issues/9844

@LLParse Can you please review?

cc @moelsayed

### CODE REVIEW

File: images/0.14.0/etcd/wrapper.go [hunk 1]
```diff
@@ -162,10 +175,52 @@ func CreateBackup(t time.Time) {
 			time.Sleep(failureInterval)
 		}
 
-		// FIXME! either get container ips (ideal) or service/stack name (dns)
-		// curl -H 'Accept: application/json' http://169.254.169.250/2016-07-29/self/stack/etcd/services/etcd/containers | jq -r .[].primary_ip
-		// only use .[].state == 'running'
-		cmd := exec.Command("etcdctl", "snapshot", "save", "--endpoints", "etcd:2379", backupDir)
+		etcdContainers, err := mClient.GetServiceContainers("etcd", "kubernetes")
```
Comment (LLParse) [inline images/0.14.0/etcd/wrapper.go]: Hard-coding the stack/service name ties this process to kubernetes. We use etcd in at least one other stack. You could instead add a call `mClient.GetSelfService()` to dynamically obtain the stack/service name.

**Comment by LLParse** [comment]
@galal-hussein Thanks for this. I just have one comment.

File: images/0.14.0/etcd/wrapper.go (hunk 1 above)
Comment (galal-hussein) [reply images/0.14.0/etcd/wrapper.go]: @LLParse yeah i will add that, one problem though, getselfservice will return `etcd-backup` and not etcd, so i can use getselfstack only

**Comment by galal-hussein** [comment]
@LLParse I added self stack, but i am not sure its the best idea, in ideal situation etcd-backup service can be used in other stacks and point etcd service in different stacks, thoughts?

**Comment by LLParse** [comment]
@galal-hussein Yes, this limits us to targeting an etcd cluster within the same stack. If you have a use case that requires this extra flexibility, you would add a template question of `Service` type. This service type can then be injected into the container via environment variable and parsed. It's up to you if you want to make the change, but I don't think we'll want multiple stacks for K8s.

See https://github.com/rancher/mesos-catalog/blob/master/images/mesos-master/entrypoint.sh#L21-L25 and https://github.com/rancher/qa-catalog/blob/master/templates/all-question-types/2/rancher-compose.yml#L53-L57 for examples.

**Comment by galal-hussein** [comment]
@LLParse I think we are good for now and no need to add extra variables to the catalog or the container for now.

**Comment by LLParse** [comment]
LGTM

**Pull request merged**